### PR TITLE
Migrating all test projects

### PR DESCRIPTION
**Pull request by jgoshi**
@piotrpMSFT @livarcocc @krwq

### CODE REVIEW

**Comment by jgoshi** [comment]
@dotnet-bot test this please

**Comment by jgoshi** [comment]
test Debian8.2 x64 Debug Build

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu x64 Release Build  please.

File: build/Microsoft.DotNet.Cli.Prepare.targets [hunk 1]
```diff
@@ -232,7 +232,7 @@
       <RestoreToolsPackagesInput Include="$(RepoRoot)/tools/**/*.csproj" />
     </ItemGroup>
   </Target>
-
+
```
Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Prepare.targets]: nit... :)

File: build/Microsoft.DotNet.Cli.Test.targets [hunk 2]
```diff
@@ -38,9 +38,10 @@
           CLIBuildDll=$(CLIBuildDll);
           Configuration=$(Configuration);
           EnvironmentVariables=$(RunTestEnvironmentVariables);
+          TestProject=%(TestProjects.ProjectPath);
           TestProjectName=%(TestProjects.OutputName);
           TestResultXmlDir=$(TestResultXmlDir);
-          ToolPath=$(Stage0PjDirectory);
+          ToolPath=$(Stage0Directory);
```
Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Test.targets]: YES!!!

File: build/Microsoft.DotNet.Cli.Test.targets [hunk 3]
```diff
@@ -71,34 +72,31 @@
...
           Outputs="%(TestProjects.BuildOutput)">
-
-    <DotNetBuildPj  Configuration="$(Configuration)"
-                    Framework="%(TestProjects.Framework)"
-                    ToolPath="$(Stage0PjDirectory)"
-                    WorkingDirectory="%(TestProjects.RelativeDir)" />
+    <DotNetBuild  Configuration="$(Configuration)"
+                  Framework="%(TestProjects.Framework)"
+                  ToolPath="$(Stage0Directory)"
+                  WorkingDirectory="%(TestProjects.RelativeDir)"
```
Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Test.targets]: Why do we need both the WorkingDirectory and the ProjectPath?

File: build/Microsoft.DotNet.Cli.Test.targets [hunk 4]
```diff
@@ -71,34 +72,31 @@
...
+                  Framework="%(TestProjects.Framework)"
+                  ToolPath="$(Stage0Directory)"
+                  WorkingDirectory="%(TestProjects.RelativeDir)" 
+                  ProjectPath="%(TestProjects.ProjectPath)" />
   </Target>
 
   <Target Name="RestoreTests"
           DependsOnTargets="CreateTestAssetPackageNuPkgs;
-                            CleanTestProjectsBinObjProjectJson;
+                            CleanTestProjectsBinObj;
```
Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Test.targets]: I can't wait to get rid of these targets [not this PR tho:)]

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs [hunk 5]
```diff
@@ -135,7 +135,7 @@ private string GetLockFilePathFromProjectLockFileProperty()
                 .AllEvaluatedProperties
                 .Where(p => p.Name.Equals("ProjectAssetsFile"))
                 .Select(p => p.EvaluatedValue)
-                .FirstOrDefault(p => File.Exists(p));
+                .FirstOrDefault(p => Path.IsPathRooted(p) && File.Exists(p));
```
Comment (piotrpMSFT) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs]: Why is this needed?

File: test/ArgumentForwardingTests/ArgumentForwardingTests.cs [hunk 6]
```diff
@@ -24,13 +23,12 @@ public class ArgumentForwardingTests : TestBase
         private string ReflectorPath { get; set; }
         private string ReflectorCmdPath { get; set; }
 
-        public static void Main()
```
Comment (piotrpMSFT) [inline test/ArgumentForwardingTests/ArgumentForwardingTests.cs]: Great :D

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs (hunk 5 above)
Comment (jgoshi) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs]: Livar pointed this out to me. When that gets evaluated it returns multiple values. One of them is the relative path for the test project.

File: build/Microsoft.DotNet.Cli.Test.targets (hunk 3 above)
Comment (jgoshi) [inline build/Microsoft.DotNet.Cli.Test.targets]: We might not. I can try taking it out (in the next PR though) if this turns green I want to get this checked in.

**Comment by jgoshi** [comment]
test Windows_NT x86 Debug Build

**Comment by jgoshi** [comment]
test Ubuntu x64 Release Build

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs (hunk 5 above)
Comment (piotrpMSFT) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs]: Sorry, I should have been more specific. Why do we want the rooted one?

File: build/Microsoft.DotNet.Cli.Test.targets (hunk 3 above)
Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Test.targets]: ok, sounds good.

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs (hunk 5 above)
Comment (jgoshi) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs]: Because without a full path we fail to open the lock file (I don't know how NuGet is resolving relative). Before my change we only had a full path. But because the test project is csproj the query now returns a relative path too. But we need to the full path to have NuGet open it. @livarcocc might have a more complete explanation :)

**Comment by jgoshi** [comment]
test RHEL7.2 x64 Release Build

**Comment by piotrpMSFT** [comment]
Wohoo!
